internal/background/queue_metrics.go

```diff
@@ -12,23 +11,21 @@ import (
 
 // jobQueueMetricLoop is a background function that runs for all workers.
 // It polls job queue statistics from Redis as well as in-flight counters.
-// It waits a random delay between 0 and sleep for better polling spread
-// since job queue size is a global metric (redis queue length).
 //
 //nolint:gosec
 func jobQueueMetricLoop(ctx context.Context, sleep time.Duration, name string) {
 	logger := zerolog.Ctx(ctx)
-
-	// spread polling intervals
-	randSleep := rand.Int63() % sleep.Milliseconds()
-	logger.Debug().Msgf("Job queue metric delay %dms", randSleep)
-	time.Sleep(time.Duration(randSleep) * time.Millisecond)
```

Comment:
Random sleep is no longer needed - this now has dedicated pod and is not running in workers.